crates/services/txpool/src/txpool.rs

```diff
@@ -391,6 +364,96 @@ where
     }
 }
 
+pub async fn check_transactions(
+    txs: &[Arc<Transaction>],
+    current_height: BlockHeight,
+    config: &Config,
+) -> Vec<anyhow::Result<CheckedTransaction>> {
+    let mut checked_txs = Vec::with_capacity(txs.len());
+
+    for tx in txs.iter() {
+        checked_txs
+            .push(check_single_tx(tx.deref().clone(), current_height, config).await);
+    }
+
+    checked_txs
+}
+
+pub async fn check_single_tx(
+    tx: Transaction,
+    current_height: BlockHeight,
+    config: &Config,
+) -> anyhow::Result<CheckedTransaction> {
+    if tx.is_mint() {
+        return Err(Error::NotSupportedTransactionType.into())
+    }
+
+    verify_tx_min_gas_price(&tx, config)?;
+
+    let tx: CheckedTransaction = if config.utxo_validation {
+        let consensus_params = &config.chain_config.transaction_parameters;
+
+        let tx = tx
+            .into_checked_basic(current_height, consensus_params)?
+            .check_signatures(&consensus_params.chain_id)?;
+
+        tx.check_predicates_async::<TokioWithRayon>(
+            consensus_params,
+            &config.chain_config.gas_costs,
+        )
+        .await?
+        .into()
+    } else {
+        tx.into_checked_basic(
+            current_height,
+            &config.chain_config.transaction_parameters,
+        )?
+        .into()
+    };
+
+    Ok(tx)
+}
+
+fn verify_tx_min_gas_price(tx: &Transaction, config: &Config) -> Result<(), Error> {
+    let price = match tx {
+        Transaction::Script(script) => script.price(),
+        Transaction::Create(create) => create.price(),
+        Transaction::Mint(_) => unreachable!(),
```

Comment:
```suggestion
        Transaction::Mint(_) => return Err(anyhow::anyhow!("Mint transactions is not supported")),
```

Reply:
I returned the appropriate error type instead `Error::NotSupportedTransactionType`